include/seqan/parallel.h

```diff
@@ -60,13 +60,13 @@
 // ----------------------------------------------------------------------------
 // Use MCSTL which is part of the GCC since version 4.3
 
-#if defined(_OPENMP) && defined(PLATFORM_GNU)
+#if defined(_OPENMP) && defined(COMPILER_GCC)
 #include <parallel/algorithm>
```

Comment:
this is definetely STDLIB_GNU

please try with clang-3.8 to see if it works. If not, we have to make it `STDLIB_GNU && COMPILER_GCC`